frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboard.tsx

```diff
@@ -259,14 +273,30 @@ class PublicOrEmbeddedDashboardInner extends Component<PublicOrEmbeddedDashboard
           })}
           loading={!dashboard}
         >
-          {() =>
-            dashboard ? (
+          {() => {
+            if (!dashboard) {
+              return null;
+            }
+
+            if (!dashboardHasCards || !tabHasCards) {
+              return (
+                <DashboardEmptyStateWithoutAddPrompt
+                  isNightMode={isNightMode}
+                />
```

Comment:
It'd be nice in the future, if there's a small comparison between the before and after, since setting up the embedding takes a bit of time. However, I've tested and previously we don't show the empty state, so this is definitely a nice touch.